### PR TITLE
cockpit: 330 -> 331

### LINKED ISSUE #367053: Regression: in cockpit supporting applications cannot be used in cockpit because libexecdir cannot be found.
## Description
Cockpit relies on multiple satellite applications to function properly however in 24.11 those no longer work because cockpit cannot find `libexecdir` to use those applications for its purposes. Purposes such as privilege escalation and adding cockpit-Applications to itself.
This all used to work in 24.05.

### WebGUI
1. Add to configuration:
```
   environment.sessionPackages = pkgs.cockpit;
   services.cockpit = {
       enable = true;
       port = 9090;
   };
```
3. open in browser `127.0.0.1:9090` or `123.456.789.x:9090` on another machine and login.
4. Eith click on `Limited access` or `Turn on Administrative privileges` and watch it either restart or give an error that `libexec/cockpit-askpass` is missing.

Also inspect the side menu there should be more there. Things like Networking, Storage, Applications and Software Updates are missing. Although mostly not relevant to nixos they should be there and its a symptom of the problem.

### Terminal
1. with cockpit installed run `cockpit-bridge`
```
cockpit.packages-WARNING: Could not detect libexecdir
1097

{
  "capabilities": {
    "explicit-superuser": true
  },
  "command": "init",
  "os-release": {
    "ANSI_COLOR": "1;34",
    "BUG_REPORT_URL": "https://github.com/NixOS/nixpkgs/issues",
    "BUILD_ID": "24.11.20241210.a0f3e10",
    "CPE_NAME": "cpe:/o:nixos:nixos:24.11",
    "DEFAULT_HOSTNAME": "nixos",
    "DOCUMENTATION_URL": "https://nixos.org/learn.html",
    "HOME_URL": "https://nixos.org/",
    "ID": "nixos",
    "ID_LIKE": "",
    "IMAGE_ID": "",
    "IMAGE_VERSION": "",
    "LOGO": "nix-snowflake",
    "NAME": "NixOS",
    "PRETTY_NAME": "NixOS 24.11 (Vicuna)",
    "SUPPORT_END": "2025-06-30",
    "SUPPORT_URL": "https://nixos.org/community.html",
    "VARIANT": "",
    "VARIANT_ID": "",
    "VENDOR_NAME": "NixOS",
    "VENDOR_URL": "https://nixos.org/",
    "VERSION": "24.11 (Vicuna)",
    "VERSION_CODENAME": "vicuna",
    "VERSION_ID": "24.11"
  },
  "version": 1,
  "packages": {
    "shell": null,
    "sosreport": null,
    "static": null,
    "metrics": null,
    "base1": null,
    "playground": null,
    "selinux": null,
    "system": null,
    "users": null
  }
}
```
note: `cockpit.packages-WARNING: Could not detect libexecdir`

3. press enter:
```
Traceback (most recent call last):
  File "/nix/store/m2l2bnrdfj14aymr82ia1jn45rxvyg6f-cockpit-328/lib/python3.12/site-packages/cockpit/protocol.py", line 130, in consume_one_frame
    length = int(data[:newline])
             ^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: b''

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/nix/store/m2l2bnrdfj14aymr82ia1jn45rxvyg6f-cockpit-328/bin/.cockpit-bridge-wrapped", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/m2l2bnrdfj14aymr82ia1jn45rxvyg6f-cockpit-328/lib/python3.12/site-packages/cockpit/bridge.py", line 315, in main
    run_async(run(args), debug=args.debug)
  File "/nix/store/m2l2bnrdfj14aymr82ia1jn45rxvyg6f-cockpit-328/lib/python3.12/site-packages/cockpit/_vendor/systemd_ctypes/event.py", line 135, in run_async
    asyncio.run(main, debug=debug)
  File "/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/zv1kaq7f1q20x62kbjv6pfjygw5jmwl6-python3-3.12.7/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/nix/store/m2l2bnrdfj14aymr82ia1jn45rxvyg6f-cockpit-328/lib/python3.12/site-packages/cockpit/bridge.py", line 166, in run
    await router.communicate()
  File "/nix/store/m2l2bnrdfj14aymr82ia1jn45rxvyg6f-cockpit-328/lib/python3.12/site-packages/cockpit/router.py", line 258, in communicate
    await self._communication_done
  File "/nix/store/m2l2bnrdfj14aymr82ia1jn45rxvyg6f-cockpit-328/lib/python3.12/site-packages/cockpit/protocol.py", line 192, in data_received
    result = self.consume_one_frame(self.buffer)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/m2l2bnrdfj14aymr82ia1jn45rxvyg6f-cockpit-328/lib/python3.12/site-packages/cockpit/protocol.py", line 132, in consume_one_frame
    raise CockpitProtocolError("frame size is not an integer") from exc
cockpit.protocol.CockpitProtocolError: frame size is not an integer
```

## Expected behavior

GUI: a password prompt at the top of the screen and Storage, Applications, etc in the side menu.

Terminal: when running cockpit `cockpit-bridge: no option specified`

## Screenshots

image

## Additional context

extra context here:
https://discourse.nixos.org/t/cockpit-dashboard-management-interface-package-does-not-work-properly/57403/8

## Metadata
```
- system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 24.11 (Vicuna), 24.11.20241210.a0f3e10`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"home-manager-24.11.tar.gz, nixos-24.11"`
 - nixpkgs: `/nix/store/va0p2i72cm2ljwm084a0g6ji41s5qnyz-source`
```

@lucas-deangelis 
---

Note for maintainers: Please tag this issue in your PR.

---

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by lucasew**
Fixes: https://github.com/NixOS/nixpkgs/issues/367053

- **cockpit: 330 -> 331**
- **tests/cockpit: each wait with its own waiter, treat /nonexistent case**

Basically, cockpit was having some FHS assuming issues. This PR bumps to the latest version and fixes them.

When merged, this should be backported because stable is also broken.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow started

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12528977197
**Extra args**: ``

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12528977197


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368886`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cockpit</li>
  </ul>
</details>

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow started

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12529066881
**Extra args**: `-p cockpit -p nixosTests.cockpit`

**Comment by lucasew** [comment]
The new iteration is just nixfmt

**Comment by lucasewd** [comment]
# [nixpkgs-reviewd](https://github.com/nixosbrasil/nixpkgs-reviewd) workflow finished

**Status and logs**: https://github.com/nixosbrasil/nixpkgs-reviewd/actions/runs/12529066881


## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 368886`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 test built:</summary>
  <ul>
    <li>nixosTests.cockpit</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cockpit</li>
  </ul>
</details>

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
Can you try adding `versionCheckHook` (see `typst/package.nix` for an example) ?

Also, for `meta`:
- remove `with lib;`,
- Add `changelog = "https://cockpit-project.org/blog/cockpit-${finalAttrs.version}.html";

I haven't looked at the module.

File: pkgs/by-name/co/cockpit/package.nix [hunk 1]
```diff
@@ -189,29 +174,31 @@ stdenv.mkDerivation rec {
...
 
-  doCheck = true;
   checkInputs = [
     bashInteractive
     cacert
     dbus
     glib-networking
     openssh
-    python3Packages.pytest
+    python3Packages.pytestCheckHook
```
[the remark this answers is not in the document]
Comment (lucasew) [reply pkgs/by-name/co/cockpit/package.nix]: Moved to nativeCheckInputs

**Review by thiagokokada — APPROVED** [review]

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Review by GaetanLepage — APPROVED** [review]
I trust you on the nixos module aspect

**Comment by lucasew** [comment]
Going to backport this as the stable version is also broken

**Comment by lucasew** [comment]
Backport PR  https://github.com/NixOS/nixpkgs/pull/369266

**Pull request merged**